### DISCUSSION TITLE
[Question] How to apply the scale factor to a JointVelocityActionCfg?

### DISCUSSION THREAD

**Discussion by Quentin2000** [isaac-sim/IsaacLab#634] [Q&A]
### Question

I am using the rsl_rl train.py script and I setup my manager_based environment config with the following ActionsCfg class:

```
@configclass
class ActionsCfg:

    joint_vel = mdp.JointVelocityActionCfg(asset_name="robot", joint_names=[".*"], scale=10.0, use_default_offset=False, debug_vis=True)
```

My current issue is that whatever the scale factor, the output actions are still between ~[-2.3;2.3] and never go above it. I tried scale factors from 0.0001 to 1000 but nothing changed. After looking at the documentation, I found out that the scale factor is applied with the process_actions() and apply_actions() methods.

```
    def apply_actions(self):
        # set joint velocity targets
        self._asset.set_joint_velocity_target(self.processed_actions, joint_ids=self._joint_ids)

    def process_actions(self, actions: torch.Tensor):
        # store the raw actions
        self._raw_actions[:] = actions
        # apply the affine transformations
        self._processed_actions = self._raw_actions * self._scale + self._offset
```

The repository doesn't show where these functions are called. Would you mind explaining how to call these?

**Comment by eferreirafilho** [comment]
When I do this:

```
self.actions.arm_action = mdp.JointVelocityActionCfg(
            asset_name="robot", joint_names=["shoulder_pan_joint", "shoulder_lift_joint", "elbow_joint", "wrist_1_joint", "wrist_2_joint", "wrist_3_joint"], scale=3.0, use_default_offset=True
        )
```

inside UR10ReachEnvCfg class in joint_pos_env_cfg.py it does scale the velocity control

**Comment by jtigue-bdai** [comment]
For the `rsl_rl/train.py` example: 

The `runner` handles the training:

The `runner` is created with an environment argument `env` which is an instance of an `RLTaskEnv` (wrapped by `RslRlVecEnvWrapper`)

The `RslRlVecEnvWrapper` calls `self.env.step(actions)` which is a call to RLTaskEnv.step()

please see: `source/extensions/omni.isaac.lab/omni/isaac/lab/envs/rl_task_env.py`:
`RLTaskEnv.step()` for details on the use of `process_action` and `apply_action`
